### PR TITLE
Fix/wrong order of requests on device disconnect

### LINKED ISSUE #2238: Wrong order of the requests that SDL sends on device disconnection
### Bug Report
On device disconnection SDL sends first UpdateAppList request, then OnAppUnregistered notification to AppLink.

##### Reproduction Steps
1. Register App
2. Disconnect transport (unplug USB or switch off BT, WiFi)

##### Expected Behavior
3. SDL sends OnAppUnregistered (with appID ofunregistered app) to AppLink
4. SDL sends UpdateAppList (with currently still registered app)

##### Observed Behavior
3. SDL sends UpdateAppList   
4. SDL sends OnAppUnregistered

##### OS & Version Information
* OS/Version: [The OS/Version you were running when the bug occurred]
* SDL Core Version: [The version of SDL Core you were using when the bug occurred]
* Testing Against: [What you tested with to observe this behavior. Proxy, App, HMI etc.]

##### Test Case, Sample Code, and / or Example App
[Paste a link to a PR, gist, or other code that exemplifies this behavior]

### PR DESCRIPTION
**Pull request by KVGrygoriev**
Fixes #2238

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
Moves clearing command list and SendOnAppUnreg notification call into separate logic unit. 
Fixes SDL behaviour when device is suddenly disconnected.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/include/application_manager/application_manager_impl.h [hunk 1]
```diff
@@ -1440,6 +1440,14 @@ class ApplicationManagerImpl
   static std::vector<std::string> ConvertRejectedParamList(
       const std::vector<std::string>& input);
 
+  /**
```
Comment (EKuliiev) [inline src/components/application_manager/include/application_manager/application_manager_impl.h]: @KVGrygoriev missed arg names in description
@param app_to_remove the application which should be removed
@param is_unexpected_disconnect specifies whether the disconnect was unexpected

Comment (KVGrygoriev) [reply src/components/application_manager/include/application_manager/application_manager_impl.h]: @EKuliiev thanks, fixed

File: src/components/application_manager/src/application_manager_impl.cc [hunk 2]
```diff
@@ -2623,12 +2630,15 @@ void ApplicationManagerImpl::UnregisterApplication(
           logger_, "There is no more SDL4 apps with device handle: " << handle);
 
       RemoveAppsWaitingForRegistration(handle);
+      ClearCommandListAndSendOnAppUnregNotificationToHMI(
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/application_manager_impl.cc]: @KVGrygoriev I really dont like functions with more than one responsibility.
I suggest you to update this code like:
```
if (!app) { 
  LOG4CXX_DEBUG( 
       logger_, "There is no more SDL4 apps with device handle: " << handle); 
       RemoveAppsWaitingForRegistration(handle); 
}

commands_holder_->Clear(app_to_remove); 
MessageHelper::SendOnAppUnregNotificationToHMI(app_to_remove, is_unexpected_disconnect, *this);  

if (!app) { 
	SendUpdateAppList(); 
}
```

Or even like this if `SendUpdateAppList` call order is not necessary for this case:
```
if (!app) { 
  LOG4CXX_DEBUG( 
       logger_, "There is no more SDL4 apps with device handle: " << handle); 
       RemoveAppsWaitingForRegistration(handle); 
       SendUpdateAppList(); 
}

commands_holder_->Clear(app_to_remove); 
MessageHelper::SendOnAppUnregNotificationToHMI(app_to_remove, is_unexpected_disconnect, *this);  
```

File: src/components/application_manager/src/application_manager_impl.cc [hunk 3]
```diff
@@ -1949,7 +1949,7 @@ bool ApplicationManagerImpl::Stop() {
   application_list_update_timer_.Stop();
   try {
     SetUnregisterAllApplicationsReason(
-      mobile_api::AppInterfaceUnregisteredReason::IGNITION_OFF);
+        mobile_api::AppInterfaceUnregisteredReason::IGNITION_OFF);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/application_manager_impl.cc]: @KVGrygoriev make this change as additional commit

Comment (KVGrygoriev) [reply src/components/application_manager/src/application_manager_impl.cc]: @LuxoftAKutsan this changes will disappear when someone fix it in develop branch

Comment (LuxoftAKutsan) [reply src/components/application_manager/src/application_manager_impl.cc]: @KVGrygoriev commit will still contain this changes and cherry pick may trigger conflicts

**Pull request merged**